doc/dev_guide.rst

```diff
@@ -203,3 +203,44 @@ is checked out this should have no side effects.
 If another custom post-checkout hook is detected on PR, then setup.py tries to append
 or update the relevant part. To prevent unwanted hook generation or update you can create
 the empty file ``.hook_ignore`` in source directory (same level as setup.py).
+
+7. Adding new methods
+---------------------
+
+With the addition of ``LazySignal`` class and its derivatives, adding methods
+that operate on the data became slightly more complicated, however we attempt
```

Comment:
And then full stop "...complicated. However, we have attempted to streamline it as much as possible."